Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,141 @@
+            }
+            
+            let installedPackagesJSONPath = try tool.fileSystem.dotSwiftPM.appending(component: "installedPackageProducts.json")
+            
+            var alreadyExisting = (try? InstalledPackageProduct.registered(tool.fileSystem)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            
+            let packageGraph = try workspace.loadPackageGraph(rootPath: packageRoot, observabilityScope: tool.observabilityScope)
+            
+            let possibleCanidates = packageGraph.rootPackages.flatMap(\.products)
+                .filter { $0.type == .executable }
+            
+            let productToInstall: Product
+            
+            switch possibleCanidates.count {
+            case 0:
+                throw StringError("No Executable Products in Package.swift.")
+            case 1:
+                productToInstall = possibleCanidates[0].underlyingProduct
+            default: // More than one, check for possible
+                guard let product, let first = possibleCanidates.first(where: { $0.name == product }) else {
+                    throw StringError("Multiple canidates found, however, no product was specified. specify a product with the --product")
+                }
+                
+                productToInstall = first.underlyingProduct
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackageProduct(url: .init(finalBinPath))
+            alreadyExisting.append(pkgInstance)
+            
+            try JSONEncoder().encode(path: installedPackagesJSONPath, fileSystem: tool.fileSystem, alreadyExisting)
+        }
+    }
+    
+    struct Uninstall: SwiftCommand {
+        
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(commandName: "experimental-uninstall", abstract: "Offers the ability to uninstall executable products of installed package products")
+        }
+        
+        @OptionGroup
+        var globalOptions: GlobalOptions
+        
+        @Argument(help: "Name of the executable to remove.")
```

Comment:
Let's make this copy consistent with the subcommand name.
```suggestion
        @Argument(help: "Name of the executable to uninstall.")
```